internal/utf16/string_test.go

```diff
@@ -0,0 +1,79 @@
+package utf16_test
+
+import (
+	"testing"
+
+	"github.com/NeowayLabs/abad/internal/utf16"
+)
+
+var S = utf16.S
+
+func TestStrIndex(t *testing.T) {
+	for _, tc := range []struct {
+		str      utf16.Str
+		sub      utf16.Str
+		index int
+		contains bool
+	}{
+		{
+			str:      S("hello world"),
+			sub:      S("h"),
+			index: 0,
+			contains: true,
+		},
+		{
+			str:      S("hello world"),
+			sub:      S("e"),
+			index: 1,
+			contains: true,
+		},
+		{
+			str:      S("hello world"),
+			sub:      S("l"),
+			index: 2,
+			contains: true,
+		},
+		{
+			str:      S("hello world"),
+			sub:      S(""),
+			index: 0,
+			contains: true,
+		},
+		{
+			str:      S("hello world"),
+			sub:      S("world"),
```

Comment:
It would be good to test only the last char (last index), like str: S("hello world") sub: S("d"). I think this is not tested..but I may be wrong (there is a lot of tests, congrats =D).

Reply:
no, lots and lots of tests cases are missing. We need to configure some coverage tool to help visualize the gaps.